.github/workflows/make_test.yaml

```
@@ -0,0 +1,46 @@
+name: Automated Tests
+
+on:
+  push:
+    paths:
+    - .github/**/*
+    - quantum/**/*
+    - tmk_core/**/*
+    - drivers/**/*
+    - tests/**/*
+    - util/**/*
+    - platforms/**/*
+    - Makefile
+    - '*.mk'
+
+jobs:
+  test:
+    name: 'make test:all'
+    runs-on: ubuntu-latest
+    container: qmkfm/base_container
+
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+
+    - uses: trilom/file-changes-action@v1.2.4
+      id: file_changes
+      with:
+        output: '\n'
+
+    - name: Print info
+      run: |
+        git rev-parse --short HEAD
+        echo ${{ github.event.pull_request.base.sha }}
+        echo '${{ steps.file_changes.outputs.files}}'
+
+    - name: Install dependencies
+      run: |
+        make git-submodule
+        python3 -m pip install -U qmk
+        python3 -m pip install -r requirements.txt
+
```

```suggestion
```

---

I'd still prefer to keep the updating of `requirements.txt` -- core changes that add anything new will hang during execution, as it asks whether or not pip should be executed during build.

---

Base container is meant to contain all the latest requirements, be it that its mostly manual right now. But i guess the CI for CLI does it.

```suggestion
    - name: Install dependencies
      run: pip3 install -r requirements.txt
```